### DISCUSSION TITLE
Block Categories

### DISCUSSION THREAD

**Discussion by X3Technology** [vanjarosoftware/Vanjaro.Platform#1468] [Ideas]
Just curious about the naming of the 'Basic' and 'Design' categories of the Blocks and what the vision is for this.

I like the idea of having different categories, but 'Basic' and 'Design' don't seem quite right.  For example, 'Signup Form' doesn't really seem like a 'design' thing, it's a functional thing.  I know it's just nomenclature, but it feels odd.

Just thought I would open the idea up for discussion.

What about:
1) Layout/design (Section, Grid, Spacer, Divider, Logo, Menu, Breadcrumb, Language, etc.)
2) Content (Heading, Text, Button, Icon, Link, List, Image, Video, Map, Carousel, etc.)
3) Admin Functionality (Signup, Sign in, User Profile, Search Input, Search Results, etc.)
4) Advanced (Custom Code (HTML), Custom JS (implement Dnn's JsInclude functionality), Custom CSS (Implement Dnn's CssInclude functionality), etc.)

It would also be great if third-party extension developers could add their own categories so they could have their own uniquely labeled group of blocks.  For example, the Mandeeps apps could all be in a 'Mandeeps' category.

**Comment by meetmandeep** [comment]
> I like the idea of having different categories, but 'Basic' and 'Design' don't seem quite right. For example, 'Signup Form' doesn't really seem like a 'design' thing, it's a functional thing. I know it's just nomenclature, but it feels odd.

Design blocks functions differently from other blocks:

- They are server rendered (similar to DNN Skin Object) via Razor Template
- They can be configured at Site Level and Instance Level

Other Blocks:

- Are rendered on client via HTML/CSS/JS
- Always Instance Level

Yes. Developers will be able to add their own categories.

**Comment by X3Technology** [comment]
Thank for the explanation.  That makes sense why there are two different groups.  I get that technical distinction, but I doubt the user cares what is behind the magic. They just see two different groups of blocks with some oddly named group label.

"Basic" and "Design" are not quite the right words for the different groups.

If you use the word "Basic", it is implied that is also an "Advanced" or "non-basic" group.  I understand that there will be "Premium" blocks in the paid version of Vanjaro, so I assume the plan is to put all of these in a "Premium" section.  But it feels weird to a person looking at Vanajro for the first time or someone on the free version.  And the word "Design" does not really fit for the Site/Instance level blocks; "Signup Form" doesn't really have anything to do with "Design", it is a functionality block.

I would suggest segmenting the blocks by use case or functional purpose.  For example, "Layout", "Content", "Site Configuration", etc.  That would feel more intuitive.

For the forthcoming "premium" blocks, perhaps use some other type of visual indicator.  A different background color, a border color, perhaps a star icon in the corner, or something else to distinguish them.
